jax/experimental/jax2tf/examples/tf_js/quickdraw/quickdraw.py

```diff
@@ -0,0 +1,149 @@
+  return QuickDrawModule().apply({'params': params}, inputs)
+
+def categorical_cross_entropy_loss(logits, labels):
+  onehot_labels = common_utils.onehot(labels, logits.shape[-1])
+  return jnp.mean(-jnp.sum(onehot_labels * jnp.log(logits), axis=1))
+
+def update(optimizer, inputs, labels):
+  def loss_fn(params):
+    logits = predict(params, inputs)
+    return categorical_cross_entropy_loss(logits, labels)
+  grad = jax.grad(loss_fn)(optimizer.target)
+  optimizer = optimizer.apply_gradient(grad)
+  return optimizer
+
+def accuracy(predict: Callable, params, dataset):
+  def top_k_classes(x, k):
+    bcast_idxs = jnp.broadcast_to(np.arange(x.shape[-1]), x.shape)
+    sorted_vals, sorted_idxs = lax.sort_key_val(x, bcast_idxs)
+    topk_idxs = (
+        lax.slice_in_dim(sorted_idxs, -k, sorted_idxs.shape[-1], axis=-1))
+    return topk_idxs
+  def _per_batch(inputs, labels):
+    logits = predict(params, inputs)
+    predicted_classes = top_k_classes(logits, 1)
+    predicted_classes = predicted_classes.reshape((predicted_classes.shape[0],))
+    return jnp.mean(predicted_classes == labels)
+  batched = [_per_batch(inputs, labels) for inputs, labels in dataset]
+  return jnp.mean(jnp.stack(batched))
+
+def train_one_epoch(optimizer, train_ds):
+  for inputs, labels in train_ds:
+    optimizer = jax.jit(update)(optimizer, inputs, labels)
+  return optimizer
+
+def init_model():
+  rng = jax.random.PRNGKey(0)
+  init_shape = jnp.ones((1, 28, 28, 1), jnp.float32)
+  initial_params = QuickDrawModule().init(rng, init_shape)["params"]
+  optimizer = flax.optim.Adam(
+      learning_rate=0.001, beta1=0.9, beta2=0.999).create(initial_params)
+  return optimizer, initial_params
+
+def train(train_ds, test_ds, classes):
+  optimizer, params = init_model()
+  for epoch in range(5):
+    start_time = time.time()
+    optimizer = train_one_epoch(optimizer, train_ds)
+    epoch_time = time.time() - start_time
+    train_acc = accuracy(predict, optimizer.target, train_ds)
+    test_acc = accuracy(predict, optimizer.target, test_ds)
+    print("Epoch {} in {:0.2f} sec".format(epoch, epoch_time))
+    print("Training set accuracy {}".format(train_acc))
+    print("Test set accuracy {}".format(test_acc))
+
+  return optimizer.target
+
+def main(*args):
+  dir_path = os.path.join(os.path.dirname(__file__), "data")
+  classes = utils.download_dataset(dir_path, NB_CLASSES)
+  assert len(classes) == NB_CLASSES, classes
+  print(f"Classes are: {classes}")
+  train_ds, test_ds = utils.load_classes(dir_path, classes)
+  flax_params = train(train_ds, test_ds, classes)
+
+  base_model_path = "/tmp/jax2tf/tf_js_quickdraw"
+  model_dir = os.path.join(base_model_path, "saved_models")
+  # the model must be converted with with_gradient set to True to be able to
+  # convert the saved model to TF.js, as "PreventGradient" is not supported
+  input_signatures = [
+    tf.TensorSpec([batch_size, 28, 28, 1]) for batch_size in [1, 256]
```

Comment:
Do we even need to save for batch_size 256? Do you know how convert_tf_saved_model handles the batch size? It converts only the first input_signature? Converts all (I'd be surprised)?

Reply:
No, it is indeed useless. It converts the one corresponding to the `serving_default` signature, which I believe would now be the first signature with the new Exporter code you pushed this morning